daemon/wrapnix.go

```diff
@@ -0,0 +1,68 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"fmt"
+	"net"
+	sys "syscall"
+)
+
+type wrapnixAddr struct {
+	net.UnixAddr
+	ucred *sys.Ucred
+}
+
+func (wa *wrapnixAddr) String() string {
+	return fmt.Sprintf("%d:%s", wa.ucred.Uid, wa.UnixAddr.String())
+}
+
+type wrapnixConn struct {
+	net.UnixConn
```

Comment:
@chipaca ah, well I almost forgot in one of my first branches that it could chase fields through the embeddings, anyway is good that this works with pointer, because copying around opaque structures of which we know little and seem usually to be passed around as pointers seems slightly risky  (random thing the original pointer could have a SetFinalizer set on it for example..., mind doesn't seem the case starting at net code but ...)